src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -253,29 +253,51 @@ private void WriteDeltaResource(object graph, ODataWriter writer, ODataSerialize
             {
                 foreach (var prop in dynamicObject.Values)
                 {
+                    IEdmProperty edmProperty = entityType?.Properties()
+                            .FirstOrDefault(p => p.Name.Equals(prop.Key));
```

Comment:
>p.Name.Equals(prop.Key) [](start = 49, length = 23)

Why simple compare the property name here?
1) does it case sensitive?
2) property alias?